src/thing/Portal.cpp

```diff
@@ -50,13 +57,44 @@ void Portal::Start() {
           }
         }
       }
+      function scan() {
+        document.getElementById('scanning').style.display='inline';
+        var xhr = new XMLHttpRequest();
+        xhr.open("GET", "/wifi/scan", true);
+        xhr.onreadystatechange = function () {
+          if (xhr.readyState==4 && xhr.status==200) {
+           load_networks(JSON.parse(xhr.responseText));
+           document.getElementById('scanning').style.display='none';
+          }
+        }
+        xhr.send();
+      }
+      function load_networks(networks) {
+        select = document.getElementById('networks');
+        select.innerHtml = '';
+        networks.sort(function(a, b) {
+          return b.rssi - a.rssi;
+        });
+        networks.forEach(function(network) {
+          option = document.createElement('option');
+          option.value = network.ssid;
+          option.text = network.ssid + ' (' + network.rssi +  ')';
+          select.add(option);
+        });
+        select.style.display='inline';
+      }
+      function set_network(select) {
+        document.getElementById('ssid').value = select[select.selectedIndex].value;
+      }
       </script>
       </head>
       <body>
       <div>Host: <input id='host'></div>
       <div>Auth: <input id='auth'></div>
       <div>Path: <input id='path'></div>
-      <div>Wifi SSID: <input id='ssid'></div>
+      <div>Wifi SSID: <input id='ssid'><button onclick='scan();'>scan</button></div>
```

Comment:
not really, what would the advantage be?

 Seems like more work, we would need a window.onload handler (might be nice to put the random "fetch_config()" call at the end of the document there too. then you would need to add an id here, fetch this button and add the onload, or just put it inline here. This seems cleaner. 

If we had a strong reason for separation, like if the JS was loaded from another file and we wanted to play with loading different JS for the same html or something like that but as this is a small file with everything inline it seems to just stick with that and mix it all together. I could be convinced otherwise though.